web/src/search/helpers.tsx

```diff
@@ -74,6 +75,50 @@ export function toggleSearchFilter(query: string, searchFilter: string): string
     return (query.substring(0, idx).trim() + ' ' + query.substring(idx + searchFilter.length).trim()).trim()
 }
 
+export function getSearchTypeFromQuery(query: string): SearchType {
+    /** RegExp to match `type:$TYPE` in any part of a query. */
+    const getTypeName = /\btype:(?<type>\w*)\b/
```

Comment:
Should this only specify valid searchTypes?

```suggestion
    const getTypeName = /\btype:(?<type>diff|commit|repo|symbol)\b/
```